File: Lista/Go/Q4.go
package main

import "fmt"

type Animal interface {
    Som() string
}

type Cachorro struct {
    Nome string
}

func (c Cachorro) Som() string {
    return "Au Au"
}

type Gato struct {
    Nome string
}

func (g Gato) Som() string {
    return "Miau"
}

func FazerSom(animal Animal) {
    fmt.Println(animal.Som())
}

func main() {
    cachorro := Cachorro{Nome: "Rex"}
    gato := Gato{Nome: "Mimi"}

    fmt.Printf("O som do %s é: ", cachorro.Nome)
    FazerSom(cachorro)

    fmt.Printf("O som da %s é: ", gato.Nome)
    FazerSom(gato)
}

File: Lista/Python/Q4.py
class Animal:
    def __init__(self, nome):
        self.nome = nome

    def som(self):
        pass

class Cachorro(Animal):
    def som(self):
        return f'{self.nome} faz: Au Au!'

class Gato(Animal):
    def som(self):
        return f'{self.nome} faz: Miau!'

cachorro = Cachorro('Rex')
gato = Gato('Mingau')

print(cachorro.som())
print(gato.som())